### PR TITLE
Add C++20 env with Cray Clang

### PR DESCRIPTION
**Pull request by havogt**

### CODE REVIEW

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

**Comment by havogt** [comment]
launch jenkins

File: jenkins/envs/daint_nvcc_gcc.sh [hunk 1]
```diff
@@ -6,7 +6,7 @@ module swap PrgEnv-cray PrgEnv-gnu
 module load cdt-cuda
 
 if [ "$build_type" != "debug" ]; then
-  module load HPX/1.5.0-CrayGNU-20.08-cuda
+  module load HPX/1.5.0-CrayGNU-20.11-cuda
```
Comment (havogt) [inline jenkins/envs/daint_nvcc_gcc.sh:9]: unrelated fix

**Review by anstaf — APPROVED** [review]
